File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/SwitchSectionShouldNotHaveTooManyStatements_CustomValue.cs
using System;

namespace Tests.Diagnostics
{
    public class Program
    {
        public Program(int myVariable)
        {
            switch (myVariable)
            {
                case 0:
                    break;
                case 1: // Noncompliant {{Reduce this switch section number of statements from 6 to at most 1, for example by extracting code into a method.}}
//              ^^^^^^^
                    Console.WriteLine("1");
                    Console.WriteLine("2");
                    Console.WriteLine("3");
                    Console.WriteLine("4");
                    Console.WriteLine("5");
                    break;
                default:
                    break;
            }
        }

        public int Test(string type)
        {
            return type switch // Compliant - FN
            {
                _ => 1
                       + 2
                       + 3
                       + 4
                       + 5
                       + 6
                       + 7
                       + 8
                       + 9
            };
        }
    }
}

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/SwitchSectionShouldNotHaveTooManyStatements_CustomValue.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text

Namespace Tests.TestCases
    Class Foo
        Public Sub Test()
            Dim number As Integer = 8

            Select Case number
                Case 1 To 5 ' Noncompliant {{Reduce this 'Case' block number of statements from 3 to at most 1, for example by extracting code into a procedure.}}
'               ^^^^^^^^^^^
                    Test()
                    Test()
                    Test()
                Case 6
                    Test()
                Case Else ' Noncompliant
                    Test()
                    Test()
                    Test()
            End Select

            Select Case number
                Case 1 To 5
                Case 6
                Case Else
            End Select

		End Sub
    End Class
End Namespace
